### PR TITLE
Remove mandatory viaIR

### PR DESCRIPTION
**Pull request by saucepoint**
## Related Issue

Mandatory viaIR sucks 💯

## Description of changes

- [x] Killed myself in Pool.sol

### CODE REVIEW

File: src/libraries/Pool.sol [hunk 1]
```diff
@@ -265,13 +265,20 @@ library Pool {
     }
 
     struct SwapParams {
+        int256 amountSpecified;
```
Comment (snreynolds) [inline src/libraries/Pool.sol:268]: does this ordering change matter?

**Review by snreynolds — APPROVED** [review]
i also do not like via-ir

File: src/libraries/Pool.sol (hunk 1 above)
Comment (saucepoint) [reply src/libraries/Pool.sol:270]: we removed the cached `exactInput = params.amountSpecified < 0`, and are now using `params.amountSpecified < 0` in place of the cached variable

making it the first element in the struct avoids the need of offset calculations

File: src/libraries/Pool.sol [hunk 2]
```diff
@@ -281,22 +283,19 @@ library Pool {
         Slot0 slot0Start = self.slot0;
         bool zeroForOne = params.zeroForOne;
 
-        uint128 liquidityStart = self.liquidity;
         uint256 protocolFee =
             zeroForOne ? slot0Start.protocolFee().getZeroForOneFee() : slot0Start.protocolFee().getOneForZeroFee();
```
Comment (hensha256) [inline src/libraries/Pool.sol:288]: `start.liquidityStart` -> `start.liquidity` ?

File: src/libraries/Pool.sol [hunk 3]
```diff
@@ -308,11 +307,10 @@ library Pool {
             swapFee = protocolFee == 0 ? lpFee : uint16(protocolFee).calculateSwapFee(lpFee);
         }
 
-        bool exactInput = params.amountSpecified < 0;
-
         // a swap fee totaling MAX_SWAP_FEE (100%) makes exact output swaps impossible since the input is entirely consumed by the fee
         if (swapFee >= SwapMath.MAX_SWAP_FEE) {
-            if (!exactInput) {
+            // if not exactInput
+            if (!(params.amountSpecified < 0)) {
```
Comment (hensha256) [inline src/libraries/Pool.sol:313]:
```suggestion
            if (params.amountSpecified >= 0) {
```
might be better?

**Review by hensha256 — APPROVED** [review]

**Pull request merged**